pkg/kubelet/eviction/eviction_manager.go

```diff
@@ -496,23 +502,56 @@ func (m *managerImpl) emptyDirLimitEviction(podStats statsapi.PodStats, pod *v1.
 			}
 		}
 	}
+
+	return false
+}
+
+func (m *managerImpl) podEphemeralStorageLimitEviction(podStats statsapi.PodStats, pod *v1.Pod) bool {
+	_, podLimits := apiv1resource.PodRequestsAndLimits(pod)
+	_, found := podLimits[v1.ResourceEphemeralStorage]
+	if !found {
+		return false
+	}
+
+	podEphemeralStorageTotalUsage := &resource.Quantity{}
+	fsStatsSet := []fsStatsType{}
+	if *m.dedicatedImageFs {
+		fsStatsSet = []fsStatsType{fsStatsLogs, fsStatsLocalVolumeSource}
```

Comment:
`fsStatsLocalVolumeSource` also contains `hostPath`, are we counting host path usage to Pod?  e.g. if we limit Pod usage to 500M and the pod mount a hostpath which is about 1Gi, are we gonna evict the pod?

Reply:
oh, my bad. I did not notice `localVolumeNames` contains other volume sources
cc @jingxu97

Reply:
Thinking further, if `hostPath` something outside rootfs, e.g. `/etc`, then do we calculate it at all?

Reply:
Yes, I think hostapath should not be counted since it is considered as persistent storage. But secrets, configumap, gitrepo, downwardapi should be counted because they are basically a wrapper of emptydir. We might need to have a separate podLocalEphemeralStorageUsage function